### PR TITLE
v3.6.1-fix huaweiagc not find platform template

### PR DESCRIPTION
**Pull request by naturalch**
Re: https://github.com/cocos/3d-tasks/issues/13623

### Changelog

* not first time to build huawei-agc，can not find buildPlatformTemplate file

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [x] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: scripts/native-pack-tool/source/utils.ts [hunk 1]
```diff
@@ -538,7 +538,7 @@ export class Paths {
         if (params.platform === 'windows') {
             this.platformTemplateDirName = params.platformParams.targetPlatform === "win32" ? "win32" : "win64";
         } else {
-            this.platformTemplateDirName = this.platform;
+            this.platformTemplateDirName = this.platform === 'huawei-agc' ? 'android' : this.platform;
```
Comment (yanOO1497) [inline scripts/native-pack-tool/source/utils.ts]: 理论上重写 validatePlatformDirectory 这个方法不是就可以了吗？这个改动是必要的吗

Comment (naturalch) [reply scripts/native-pack-tool/source/utils.ts]: 是必要的，其他方法中用到 `this.paths.platformTemplateDirInPrj` 这个值就是从 `this.palatformTemplateName` 获取的，所以我统一复用 ‘Android’ 了

Comment (yanOO1497) [inline scripts/native-pack-tool/source/utils.ts]: 这里其实有点不规范，理论上这些平台判断应该移到平台的构建 tool 上，而不是在这个公用的地方判断。

**Review by yanOO1497 — APPROVED** [review]

File: scripts/native-pack-tool/source/utils.ts [hunk 2]
```diff
@@ -591,4 +591,8 @@ export class Paths {
     get nativePrjDir() {
         return ps.join(this.buildDir, 'proj');
     }
+
+    setPlatformTemplateDirName(platform: string) {
```
Comment (yanOO1497) [inline scripts/native-pack-tool/source/utils.ts]: 这个方法好像没有地方使用？ 其实 platformTemplateDirName  直接设成 public 就可以了

Comment (naturalch) [reply scripts/native-pack-tool/source/utils.ts]: > 理论上重写 validatePlatformDirectory 这个方法不是就可以了吗？这个改动是必要的吗

对的，我统一修改 `this.palatformTemplateName` 是想着 project/native/engine/huawei-agc 目录也复用 android 的，但其实只要复用 engine/template/android 就好

> 这个方法好像没有地方使用？ 其实 platformTemplateDirName 直接设成 public 就可以了 

这个方法没用，已删除

**Review by yanOO1497 — APPROVED** [review]

**Review by VisualSJ — APPROVED** [review]

**Pull request merged**